### PR TITLE
Ban Module

### LINKED ISSUE #217: Bans aren't transferred from PHPBB3
As far as I can see, there's no code to handle that at all, so I guess this is a feature request.

### PR DESCRIPTION
**Pull request by veryard**
- phpBB 3 Ban Module

### CODE REVIEW

File: boards/phpbb3.php
Comment (euantorano) [inline boards/phpbb3.php:8]: I assume the large diff here is due to some automatic formatting from an IDE/editor?

File: resources/modules/bans.php
Comment (euantorano) [inline resources/modules/bans.php]: This should be `Converter_Module_Bans` I believe:

```suggestion
abstract class Converter_Module_Bans extends Converter_Module
```

File: boards/phpbb3/bans.php
Comment (euantorano) [inline boards/phpbb3/bans.php]: This copyright year should be 2019:

```suggestion
 * Copyright 2019 MyBB Group, All Rights Reserved
```

**Review by euantorano — CHANGES_REQUESTED** [review]
This looks good to me, just some minor comments that I've spotted so far.

I haven't tested this at all yet, to do so I will need to spin up a PHPBB 3 install and ban some users. I'll try and get a chance to do that this weekend.

File: boards/phpbb3.php
Comment (veryard) [reply boards/phpbb3.php:8]: Yeah, recently started using PHPStorm.

File: resources/modules/bans.php
Comment (veryard) [reply resources/modules/bans.php]: Whoops! Had it named `Converted` in both cases, renamed for consistency.

File: boards/phpbb3/bans.php
Comment (veryard) [reply boards/phpbb3/bans.php]: Updated my files to 2019.

**Comment by burner1024** [comment]
Just tried this.
1) All bans end today (well, at least permanent ones, I only have those).
2) Ban reason is missing.
3) Ban source is always "MyBB Engine".
4) Several "Guest" accounts somehow creeped in? Maybe those are deleted users?

Captura de pantalla de 2019-03-22 21-20-09

I don't mean to be picky, of course, just trying to help too.

**Comment by veryard** [comment]
> Just tried this.
> 
> 1. All bans end today (well, at least permanent ones, I only have those).
> 2. Ban reason is missing.
> 3. Ban source is always "MyBB Engine".
> 4. Several "Guest" accounts somehow creeped in? Maybe those are deleted users?
> 
> Captura de pantalla de 2019-03-22 21-20-09
> 
> I don't mean to be picky, of course, just trying to help too.

Thanks for the feedback:
1. Will have a look, MyBB has a weird dateline field that i'll need to code from phpBB.
2. phpBB has two reasons, one that is given and one that is shown, at the moment i'm just pulling the `ban_reason` not the `ban_give_reason` (I'll make it pull the `ban_give_reason` if the initial one is empty)
3. I don't think phpBB store the user id against the ban when banned? (Would need to pull from the `phpbb_log` but these can be cleared so im not to sure.
4. I'm not sure about this, I only had 2 users and they both came through no guests.

I'll get to everything on Sunday.

Thanks,
Brad.

File: boards/phpbb3/bans.php
Comment (euantorano) [reply boards/phpbb3/bans.php]: Perfect, thanks!

**Comment by euantorano** [comment]
I'll keep this as requiring changes until I've had a chance to test, then approve it if it looks like everything works. Thanks for the contribution 😄

**Comment by burner1024** [comment]
2. I definitely have an account with both reasons filled in, still comes up empty in MyBB.
3. I think you're right about the logs. It would be helpful to pull the source from there, but it's not crucial.
4. Yes, it appears that deleted users can leave lingering bans, db integrity is not enforced. Can probably just skip these:
```
select count(ban_id) from phpbb_banlist where ban_userid not in (select user_id from phpbb_users);
+---------------+
| count(ban_id) |
+---------------+
|             8 |
+---------------+
```

5. Also, there are IP and email bans, ideally they should be transferred as well.

**Comment by veryard** [comment]
Hello,

Have made changes: 
- Reason will pull from `ban_give_reason` if `ban_reason` is empty
- Deleted users or users that do not exist in the users table are ignored.
- `bantime` is now generated based from the phpBB fields `ban_start` & `ban_end`, treats bans less than one day as a one day ban. (phpBB allows users to be banned for 30 mins, 1 hour etc etc.)



> 5. Also, there are IP and email bans, ideally they should be transferred as well.

I looked into this, looks like MyBB stores these bans in the `banfilters` table which is a bit odd considering it all could be done from one table. I'll have to extend the code to do this some where down the line.

**Comment by euantorano** [comment]
I’d be happy with email/IP bans being added at a later date, I think having just standard band would be a big enough improvement on its own.

**Comment by veryard** [comment]
Let me know if you need me to fix/update anything.

Brad.

**Comment by burner1024** [comment]
What if the text contains valid tags, though? Code snippets, etc.

**Comment by veryard** [comment]
> What if the text contains valid tags, though? Code snippets, etc.

Sorry, I don't understand what you're saying?

**Comment by burner1024** [comment]
err... that was for the pull for issue #223

**Comment by veryard** [comment]
> err... that was for the pull for issue #223
I see, well phpBB uses BBCodes aswell, so they are code snippets are inside the `[code]` BBCode.

**Review by euantorano — APPROVED** [review]

**Pull request merged**
